pkg/core/candidate/broker.go

```diff
@@ -171,12 +174,13 @@ func (b *Broker) requestCandidate(hash []byte) (message.Candidate, error) {
 
 func (b *Broker) Validate(m message.Message) error {
 	cm := m.Payload().(message.Candidate)
-	b.lock.RLock()
-	defer b.lock.RUnlock()
-	if _, ok := b.queue[string(cm.Block.Header.Hash)]; ok {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	if _, ok := b.republished[string(cm.Block.Header.Hash)]; ok {
 		return republisher.DuplicatePayloadError
 	}
 
+	b.republished[string(cm.Block.Header.Hash)] = struct{}{}
```

Comment:
any reason why this is not simply a `delete(b.republished, string(cm.Block.Header.Hash))`?

Reply:
Why would I delete something that's not there in the first place?

Reply:
ah yes, completely misread the code. Was mislead by the fact that a `map` is used instead of a `hashset`